src/libUtils/DataConversion.cpp

```diff
@@ -81,6 +81,36 @@ bool DataConversion::Uint8VecToHexStr(const bytes& hex_vec, string& str) {
   return true;
 }
 
+bool DataConversion::HexStringsSame(const std::string& a,
```

Comment:
This code may not be needed if we rely on `FixedHash<>` `operator==`.
So instead of comparing strings, you can do sth like this:
```TransactionHash hash_1{a};
TransactionHash hash_2{b};
if(a == b) {}```

Reply:
that will work only if they are the right length, the casing is the same, and there are no '0x' to be seen - so I thought my function more general. I can switch it to your way if you like(?)

Reply:
Thanks for explaining. But in the case here you use this method for comparing two tx hashes - which are supposed to be of the same length (if not, then one hash may be padded with zeros and would return false for equality anyway), different casing is ok (since hex representation doesn't matter as long underlying bytes are the same). What do you think?

Reply:
The code breaks not at this point but lower down when it checks the size of the string coming in (it throws). It is ok when you remove this checking code, so I have taken off the '0x' stripping and removed the checking code. Hashes that are too long or short fail in the same way as hashes that are not found, it seems, so all is well.

Reply:
Yes, it is better this way - you can find the change in this most recent commit